### ISSUE TITLE
在特定领域上微调运行脚本报错：finetune.py: error: the following arguments are required: --output_dir

### ISSUE THREAD

**Issue by snow-like-kk** [zjunlp/DeepKE#510]
bash文件的内容如下：
output_dir='lora/oneke_continue'
mkdir -p ${output_dir}
echo "Output directory is set to: ${output_dir}"
CUDA_VISIBLE_DEVICES="0,1,2,3" torchrun --nproc_per_node=4 --master_port=1287 src/finetune.py \
    --output_dir=${output_dir} \
    --do_train --do_eval \
    --overwrite_output_dir \
    --model_name_or_path '/home/xubo/xcx/NER/OneKE' \
    --stage 'sft' \
    --model_name 'llama' \
    --template 'llama2_zh' \
    --train_file '/home/xubo/xcx/NER/data/train.json' \
    --output_dir=${output_dir} \
    --per_device_train_batch_size 2 \
    --per_device_eval_batch_size 2 \
    --gradient_accumulation_steps 4 \
    --preprocessing_num_workers 16 \
    --num_train_epochs 10 \
    --learning_rate 5e-5 \
    --max_grad_norm 0.5 \
    --optim "adamw_torch" \
    --max_source_length 400 \
    --cutoff_len 700 \
    --max_target_length 300 \
    --evaluation_strategy "epoch" \
    --save_strategy "epoch" \
    --save_total_limit 10 \
    --lora_r 64 \
    --lora_alpha 64 \
    --lora_dropout 0.05 \
    --bf16 \
    --bits 4

**Comment by guihonghao** [comment]
你好，注意到脚本中已给出output_dir参数。但是，命令行调用需要在每行的末尾增加 / 符号，例如：
```bash
output_dir='lora/oneke_continue'
mkdir -p ${output_dir}
echo "Output directory is set to: ${output_dir}"
CUDA_VISIBLE_DEVICES="0,1,2,3" torchrun --nproc_per_node=4 --master_port=1287 src/finetune.py \
--output_dir=${output_dir} \
--do_train --do_eval \
--overwrite_output_dir \
--model_name_or_path '/home/xubo/xcx/NER/OneKE' \
...
```

**Comment by snow-like-kk** [comment]
> 你好，注意到脚本中已给出output_dir参数。但是，命令行调用需要在每行的末尾增加 \ 符号，例如：
> 
> ```shell
> output_dir='lora/oneke_continue'
> mkdir -p ${output_dir}
> echo "Output directory is set to: ${output_dir}"
> CUDA_VISIBLE_DEVICES="0,1,2,3" torchrun --nproc_per_node=4 --master_port=1287 src/finetune.py \
> --output_dir=${output_dir} \
> --do_train --do_eval \
> --overwrite_output_dir \
> --model_name_or_path '/home/xubo/xcx/NER/OneKE' \
> ...
> ```

![image](https://github.com/zjunlp/DeepKE/assets/132073660/d85b7e79-e141-43e9-8cf5-5d811344898a)
有写 \ 符号的，具体报错如下：
![image](https://github.com/zjunlp/DeepKE/assets/132073660/4643c21c-05e2-496d-a56a-9d7b693f30a1)

**Comment by guihonghao** [comment]
![image](https://github.com/zjunlp/DeepKE/assets/53286894/fa416aa5-93c7-41d1-a74e-1a24f7be4c72)
参考之前的issue，可能是多余空格问题导致的问题。

**Comment by snow-like-kk** [comment]
没有修改过原本的格式，仍然不行

**Comment by guihonghao** [comment]
<img width="908" alt="image" src="https://github.com/zjunlp/DeepKE/assets/53286894/07baaf6b-c735-4687-a905-c7fc7b3723b1">
有两个重复的参数，或许可以删去一个

**Comment by zxlzr** [comment]
请问您的问题解决了吗

**Comment by jingrangege** [comment]
C:\Users\workstation\.conda\envs\qwen-vl\python.exe C:\Users\workstation\Desktop\Qwen-VL-master\finetune.py C:\Users\workstation\Desktop\Qwen-VL-master\output_dir 
C:\Users\workstation\.conda\envs\qwen-vl\lib\site-packages\deepspeed\runtime\zero\linear.py:44: FutureWarning: `torch.cuda.amp.custom_fwd(args...)` is deprecated. Please use `torch.amp.custom_fwd(args..., device_type='cuda')` instead.
  def forward(ctx, input, weight, bias=None):
C:\Users\workstation\.conda\envs\qwen-vl\lib\site-packages\deepspeed\runtime\zero\linear.py:70: FutureWarning: `torch.cuda.amp.custom_bwd(args...)` is deprecated. Please use `torch.amp.custom_bwd(args..., device_type='cuda')` instead.
  def backward(ctx, grad_output):
C:\Users\workstation\.conda\envs\qwen-vl\lib\site-packages\transformers\utils\generic.py:482: FutureWarning: `torch.utils._pytree._register_pytree_node` is deprecated. Please use `torch.utils._pytree.register_pytree_node` instead.
  _torch_pytree._register_pytree_node(
C:\Users\workstation\.conda\envs\qwen-vl\lib\site-packages\transformers\utils\generic.py:339: FutureWarning: `torch.utils._pytree._register_pytree_node` is deprecated. Please use `torch.utils._pytree.register_pytree_node` instead.
  _torch_pytree._register_pytree_node(
C:\Users\workstation\.conda\envs\qwen-vl\lib\site-packages\transformers\deepspeed.py:24: FutureWarning: transformers.deepspeed module is deprecated and will be removed in a future version. Please import deepspeed modules directly from transformers.integrations
  warnings.warn(
usage: finetune.py [-h] [--model_name_or_path MODEL_NAME_OR_PATH]
                   [--data_path DATA_PATH] [--eval_data_path EVAL_DATA_PATH]
                   [--lazy_preprocess [LAZY_PREPROCESS]] --output_dir
                   OUTPUT_DIR [--overwrite_output_dir [OVERWRITE_OUTPUT_DIR]]
                   [--do_train [DO_TRAIN]] [--do_eval [DO_EVAL]]
                   [--do_predict [DO_PREDICT]]
                   [--eval_strategy {no,steps,epoch}]
                   [--prediction_loss_only [PREDICTION_LOSS_ONLY]]
                   [--per_device_train_batch_size PER_DEVICE_TRAIN_BATCH_SIZE]
                   [--per_device_eval_batch_size PER_DEVICE_EVAL_BATCH_SIZE]
                   [--per_gpu_train_batch_size PER_GPU_TRAIN_BATCH_SIZE]
                   [--per_gpu_eval_batch_size PER_GPU_EVAL_BATCH_SIZE]
                   [--gradient_accumulation_steps GRADIENT_ACCUMULATION_STEPS]
                   [--eval_accumulation_steps EVAL_ACCUMULATION_STEPS]
                   [--eval_delay EVAL_DELAY]
                   [--torch_empty_cache_steps TORCH_EMPTY_CACHE_STEPS]
                   [--learning_rate LEARNING_RATE]
                   [--weight_decay WEIGHT_DECAY] [--adam_beta1 ADAM_BETA1]
                   [--adam_beta2 ADAM_BETA2] [--adam_epsilon ADAM_EPSILON]
                   [--max_grad_norm MAX_GRAD_NORM]
                   [--num_train_epochs NUM_TRAIN_EPOCHS]
                   [--max_steps MAX_STEPS]
                   [--lr_scheduler_type {linear,cosine,cosine_with_restarts,polynomial,constant,constant_with_warmup,inverse_sqrt,reduce_lr_on_plateau,cosine_with_min_lr,warmup_stable_decay}]
                   [--lr_scheduler_kwargs LR_SCHEDULER_KWARGS]
                   [--warmup_ratio WARMUP_RATIO] [--warmup_steps WARMUP_STEPS]
                   [--log_level {detail,debug,info,warning,error,critical,passive}]
                   [--log_level_replica {detail,debug,info,warning,error,critical,passive}]
                   [--log_on_each_node [LOG_ON_EACH_NODE]]
                   [--no_log_on_each_node] [--logging_dir LOGGING_DIR]
                   [--logging_strategy {no,steps,epoch}]
                   [--logging_first_step [LOGGING_FIRST_STEP]]
                   [--logging_steps LOGGING_STEPS]
                   [--logging_nan_inf_filter [LOGGING_NAN_INF_FILTER]]
                   [--no_logging_nan_inf_filter]
                   [--save_strategy {no,steps,epoch}]
                   [--save_steps SAVE_STEPS]
                   [--save_total_limit SAVE_TOTAL_LIMIT]
                   [--save_safetensors [SAVE_SAFETENSORS]]
                   [--no_save_safetensors]
                   [--save_on_each_node [SAVE_ON_EACH_NODE]]
                   [--save_only_model [SAVE_ONLY_MODEL]]
                   [--restore_callback_states_from_checkpoint [RESTORE_CALLBACK_STATES_FROM_CHECKPOINT]]
                   [--no_cuda [NO_CUDA]] [--use_cpu [USE_CPU]]
                   [--use_mps_device [USE_MPS_DEVICE]] [--seed SEED]
                   [--data_seed DATA_SEED] [--jit_mode_eval [JIT_MODE_EVAL]]
                   [--use_ipex [USE_IPEX]] [--bf16 [BF16]] [--fp16 [FP16]]
                   [--fp16_opt_level FP16_OPT_LEVEL]
                   [--half_precision_backend {auto,apex,cpu_amp}]
                   [--bf16_full_eval [BF16_FULL_EVAL]]
                   [--fp16_full_eval [FP16_FULL_EVAL]] [--tf32 TF32]
                   [--local_rank LOCAL_RANK]
                   [--ddp_backend {nccl,gloo,mpi,ccl,hccl,cncl}]
                   [--tpu_num_cores TPU_NUM_CORES]
                   [--tpu_metrics_debug [TPU_METRICS_DEBUG]]
                   [--debug DEBUG [DEBUG ...]]
                   [--dataloader_drop_last [DATALOADER_DROP_LAST]]
                   [--eval_steps EVAL_STEPS]
                   [--dataloader_num_workers DATALOADER_NUM_WORKERS]
                   [--dataloader_prefetch_factor DATALOADER_PREFETCH_FACTOR]
                   [--past_index PAST_INDEX] [--run_name RUN_NAME]
                   [--disable_tqdm DISABLE_TQDM]
                   [--remove_unused_columns [REMOVE_UNUSED_COLUMNS]]
                   [--no_remove_unused_columns]
                   [--label_names LABEL_NAMES [LABEL_NAMES ...]]
                   [--load_best_model_at_end [LOAD_BEST_MODEL_AT_END]]
                   [--metric_for_best_model METRIC_FOR_BEST_MODEL]
                   [--greater_is_better GREATER_IS_BETTER]
                   [--ignore_data_skip [IGNORE_DATA_SKIP]] [--fsdp FSDP]
                   [--fsdp_min_num_params FSDP_MIN_NUM_PARAMS]
                   [--fsdp_config FSDP_CONFIG]
                   [--fsdp_transformer_layer_cls_to_wrap FSDP_TRANSFORMER_LAYER_CLS_TO_WRAP]
                   [--accelerator_config ACCELERATOR_CONFIG]
                   [--deepspeed DEEPSPEED]
                   [--label_smoothing_factor LABEL_SMOOTHING_FACTOR]
                   [--optim OPTIM] [--optim_args OPTIM_ARGS]
                   [--adafactor [ADAFACTOR]]
                   [--group_by_length [GROUP_BY_LENGTH]]
                   [--length_column_name LENGTH_COLUMN_NAME]
                   [--report_to REPORT_TO]
                   [--ddp_find_unused_parameters DDP_FIND_UNUSED_PARAMETERS]
                   [--ddp_bucket_cap_mb DDP_BUCKET_CAP_MB]
                   [--ddp_broadcast_buffers DDP_BROADCAST_BUFFERS]
                   [--dataloader_pin_memory [DATALOADER_PIN_MEMORY]]
                   [--no_dataloader_pin_memory]
                   [--dataloader_persistent_workers [DATALOADER_PERSISTENT_WORKERS]]
                   [--skip_memory_metrics [SKIP_MEMORY_METRICS]]
                   [--no_skip_memory_metrics]
                   [--use_legacy_prediction_loop [USE_LEGACY_PREDICTION_LOOP]]
                   [--push_to_hub [PUSH_TO_HUB]]
                   [--resume_from_checkpoint RESUME_FROM_CHECKPOINT]
                   [--hub_model_id HUB_MODEL_ID]
                   [--hub_strategy {end,every_save,checkpoint,all_checkpoints}]
                   [--hub_token HUB_TOKEN]
                   [--hub_private_repo [HUB_PRIVATE_REPO]]
                   [--hub_always_push [HUB_ALWAYS_PUSH]]
                   [--gradient_checkpointing [GRADIENT_CHECKPOINTING]]
                   [--gradient_checkpointing_kwargs GRADIENT_CHECKPOINTING_KWARGS]
                   [--include_inputs_for_metrics [INCLUDE_INPUTS_FOR_METRICS]]
                   [--eval_do_concat_batches [EVAL_DO_CONCAT_BATCHES]]
                   [--no_eval_do_concat_batches]
                   [--fp16_backend {auto,apex,cpu_amp}]
                   [--evaluation_strategy {no,steps,epoch}]
                   [--push_to_hub_model_id PUSH_TO_HUB_MODEL_ID]
                   [--push_to_hub_organization PUSH_TO_HUB_ORGANIZATION]
                   [--push_to_hub_token PUSH_TO_HUB_TOKEN]
                   [--mp_parameters MP_PARAMETERS]
                   [--auto_find_batch_size [AUTO_FIND_BATCH_SIZE]]
                   [--full_determinism [FULL_DETERMINISM]]
                   [--torchdynamo TORCHDYNAMO] [--ray_scope RAY_SCOPE]
                   [--ddp_timeout DDP_TIMEOUT]
                   [--torch_compile [TORCH_COMPILE]]
                   [--torch_compile_backend TORCH_COMPILE_BACKEND]
                   [--torch_compile_mode TORCH_COMPILE_MODE]
                   [--dispatch_batches DISPATCH_BATCHES]
                   [--split_batches SPLIT_BATCHES]
                   [--include_tokens_per_second [INCLUDE_TOKENS_PER_SECOND]]
                   [--include_num_input_tokens_seen [INCLUDE_NUM_INPUT_TOKENS_SEEN]]
                   [--neftune_noise_alpha NEFTUNE_NOISE_ALPHA]
                   [--optim_target_modules OPTIM_TARGET_MODULES]
                   [--batch_eval_metrics [BATCH_EVAL_METRICS]]
                   [--eval_on_start [EVAL_ON_START]]
                   [--eval_use_gather_object [EVAL_USE_GATHER_OBJECT]]
                   [--cache_dir CACHE_DIR]
                   [--model_max_length MODEL_MAX_LENGTH]
                   [--use_lora [USE_LORA]] [--no_use_lora]
                   [--fix_vit [FIX_VIT]] [--no_fix_vit] [--lora_r LORA_R]
                   [--lora_alpha LORA_ALPHA] [--lora_dropout LORA_DROPOUT]
                   [--lora_target_modules LORA_TARGET_MODULES [LORA_TARGET_MODULES ...]]
                   [--lora_weight_path LORA_WEIGHT_PATH]
                   [--lora_bias LORA_BIAS] [--q_lora [Q_LORA]]
finetune.py: error: the following arguments are required: --output_dir

进程已结束，退出代码为 2
可以帮忙解决一下吗

**Comment by guihonghao** [comment]
设置--output_dir参数
